docs/technical/tests/frontend/components.rst

```diff
@@ -0,0 +1,37 @@
+Component Tests
+===============
+
+When it comes to testing React components, we need a way to test components in insolation without needing to mount the entire application. Many of our current React components are tested with `Enzyme <https://airbnb.io/enzyme/>`_, but more recently, our tests have started to use `React Testing Library <https://testing-library.com/docs/react-testing-library/intro>`_ instead. (See `this blog post by Kent Dodds for more information <https://kentcdodds.com/blog/introducing-the-react-testing-library>`_.)
+
+The main reason to adopdt React Testing Library was to avoid testing too much implementation details, especially for React components. You want your tests to be maintanable and don't break and slow you down when you refactor components. With React testing Library we're testing the actual DOM and not React-specific implementation details, like props or state.
```

Comment:
- `adopdt` -> `adopt`
- `maintanable` -> `maintainable`
- `React testing Library` -> capitalize T `React Testing Library`
- put `props` and `state` inside code markup